MekHQ/src/mekhq/campaign/personnel/autoAwards/AutoAwardsController.java

```diff
@@ -787,11 +787,17 @@ private Map<Integer, List<Object>> ScenarioKillAwardsManager(List<UUID> personne
         for (UUID person : personnel) {
             Map<Integer, List<Object>> data;
 
-            List<Kill> kills = campaign.getKillsFor(campaign.getPerson(person).getId());
-            kills.removeIf(kill -> kill.getScenarioId() != scenarioId);
+            List<Kill> allKills = campaign.getKillsFor(campaign.getPerson(person).getId());
+            List<Kill> scenarioKills = new ArrayList<>();
+
+            for (Kill kill : allKills) {
```

Comment:
This can get a little bit inefficient for long-serving pilots with hundreds or thousands of kills. Any thoughts on optimizing it?

Reply:
A little rework might make it so that only kills from the scenario just concluded are passed in, which would remove the vast majority of the necessary filtering. Will depend on how scenario kills are distributed during the post-scenario handling in mhq.

Reply:
Was an easy fix. Thanks for mentioning this